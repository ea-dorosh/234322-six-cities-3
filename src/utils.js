export const getRandomNumber = (min, max) => {
  return Math.floor(min + Math.random() * (max + 1 - min));
};

export const getRandomElement = (array) => {
  return array[getRandomNumber(0, array.length - 1)];
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getCities = (array) => {
  const cities = array.map((item) => item.city);
  const citiesByName = cities.reduce((acc, city) => {
    acc[city.name] = city; return acc;
  }, {});

  return Object.values(citiesByName);
};

export const getOffersByCity = (array, city) => {
  return array.filter((item) => item.city.name === city);
};

export const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

export const CITIES = `cities`;

const ratingToStarMap = {1: 20, 2: 40, 3: 60, 4: 80, 5: 100};
export const ratingToStar = (rating) => ratingToStarMap[Math.round(rating)] || Math.round(rating);
