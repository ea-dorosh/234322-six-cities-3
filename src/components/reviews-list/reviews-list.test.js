import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list.jsx";
import {getRandomNumber} from "../../utils";

const review = {
  text: `some information about offer, some information about offer`,
  rating: parseFloat(`${getRandomNumber(2, 4)}.${getRandomNumber(0, 9)}`),
  name: `Kostya`,
  date: `2017-05-26`,
  id: 2,
};

it(`Should ReviewsList render correctly`, () => {
  const tree = renderer
    .create(<ReviewsList
      review={review}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

