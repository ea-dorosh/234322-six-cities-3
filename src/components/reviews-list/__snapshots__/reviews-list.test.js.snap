// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ReviewsList render correctly 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Konstantin
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        some information about offer, some information about offer
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        April 2019
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Kostya
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        some information about offer, some information about offer
      </p>
      <time
        className="reviews__time"
        dateTime="2017-05-26"
      >
        May 2017
      </time>
    </div>
  </li>
</ul>
`;
